src/shared/codelens/typescriptCodeLensProvider.ts

```diff
@@ -280,12 +289,21 @@ class LocalLambdaRunner {
             this.localInvokeArgs.handlerName
         ).replace('\\', '/')
 
-        await new SamTemplateGenerator()
+        const existingTemplateResource =
+            await this.getResourceForHandlerFromTemplate(relativeFunctionHandler, await detectLocalLambdas())
+
+        const newTemplate = new SamTemplateGenerator()
             .withCodeUri(rootCodeFolder)
             .withFunctionHandler(relativeFunctionHandler)
             .withResourceName(LocalLambdaRunner.TEMPLATE_RESOURCE_NAME)
             .withRuntime(this.runtime)
-            .generate(inputTemplatePath)
+
+        if (!!existingTemplateResource && !!existingTemplateResource.Properties &&
```

Comment:
Nit: `!!` is not necessary when the expression is already being treated as a boolean (there are many places that we use a `!!` where it is not necessary, but we're trying to avoid it in new code).

`!!` is only necessary when the type of the expression is inferred, i.e.:

```typescript
function isNonEmpty(str: string): boolean {
    return str // wrong, returns the actual value of str
    return !!str // correct, returns `true` or `false`
}
```